openproject/client.py

```diff
@@ -0,0 +1,170 @@
+                    (resp[key] for key in error_message_keys if key in resp), None
+                )
+                if error_message:
+                    error = f"Message: {resp.get('message')} , Error details: {resp.get('errors')}, {resp.get('data')}"
+                    raise APIError(error, response)
+
+            raise APIError(error, response)
+
+    def _send_request(
+        self, method: str, endpoint: str, params=None, data=None, **kwargs
+    ) -> httpx.Response:
+        url = f"{self.base_url}/api/{self.api_version}/{endpoint}"
+        with httpx.Client(auth=("apikey", self.api_token)) as client:
+            headers = {"Content-Type": "application/json"}
+            response = client.request(
+                method, url, params=params, json=data, headers=headers, **kwargs
+            )
+            return self._handle_response(response)
+
+
+class SubClient(Client):
+    def __init__(self, client: Client):
+        self.client = client
+
+
+class WorkPackages(SubClient):
+    _args_api_mapping = {
+        "_links": "_links",
+        "subject": "subject",
+        "description": "description",
+        "schedule_manually": "scheduleManually",
+        "readonly": "readonly",
+        "start_date": "startDate",
+        "due_date": "dueDate",
+        "derived_start_date": "derivedStartDate",
+        "derived_due_date": "derivedDueDate",
+        "estimated_time": "estimatedTime",
+        "derived_estimated_time": "derivedEstimatedTime",
+        "percentage_done": "percentageDone",
+        "custom_field_1": "customField1",
+        "custom_field_2": "customField2",
+        "created_at": "createdAt",
+        "updated_at": "updatedAt",
+    }
+
+    def list(self):
+        return self.client._send_request("GET", "/work_packages")
+
+    def view(self, id: int):
+        return self.client._send_request("GET", f"/work_packages/{id}")
+
+    def _process_data(self, data):
+        data = {
+            k: v
+            for k, v in data.items()
+            if v is not None and k != "self" and k != "kwargs"
+        }
+        return data
+
+    def create(
+        self,
+        _links: dict = None,
+        subject: str = None,
+        description: str = None,
+        schedule_manually: bool = None,
+        readonly: bool = None,
+        start_date: str = None,
+        due_date: str = None,
+        derived_start_date: str = None,
+        derived_due_date: str = None,
+        estimated_time: str = None,
+        derived_estimated_time: str = None,
+        percentage_done: int = None,
+        custom_field_1: str = None,
+        custom_field_2: str = None,
+        created_at: str = None,
+        updated_at: str = None,
+        **kwargs,
+    ):
+        if _links is None and kwargs:
+            _links = kwargs
+
+        data = {
+            self._args_api_mapping[k]: v
+            for k, v in locals().items()
+            if v is not None and k != "self" and k != "kwargs"
+        }
```

Comment:
This is not correct, as `locals` will return all local variables. Instead, I suggest we only accept `kwargs` but make it a `TypedDict`[^1].

Also, since we duplicate the cleaning of variables, let's offload it to a method (e.g. `_api_payload_from_kwargs`)

[^1]: https://docs.python.org/3/library/typing.html#typing.TypedDict